clippy_lints/src/unnested_or_patterns.rs

```diff
@@ -120,18 +120,24 @@ fn lint_unnested_or_patterns(cx: &EarlyContext<'_>, pat: &Pat) {
 
 /// Remove all `(p)` patterns in `pat`.
 fn remove_all_parens(pat: &mut P<Pat>) {
-    struct Visitor;
+    #[derive(Default)]
+    struct Visitor {
+        /// If is not in the outer most pattern
+        is_inner: bool,
+    }
+
     impl MutVisitor for Visitor {
         fn visit_pat(&mut self, pat: &mut P<Pat>) {
+            let is_inner = mem::replace(&mut self.is_inner, true);
```

Comment:
Using `mem::replace` here looks a bit overkill to me.
```suggestion
            *self.is_inner = true;
```

Reply:
Not really a overkill I think, but anyway I've replaced with normal ways.